### PR TITLE
[SYCL][NewPM] Use new pass manager for LowerWGScope pass execution

### PR DESCRIPTION
**Pull request by mlychkov**
Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

**Review by smanna12 — COMMENTED** [review]
I am not familiar with this functionality, so I leave the review to @bader / @AlexeySachkov

**Comment by bader** [comment]
@AlexeySachkov, ping.

File: clang/lib/CodeGen/CodeGenAction.cpp [hunk 1]
```diff
@@ -332,9 +332,24 @@ namespace clang {
...
+        CGSCCAnalysisManager CGAM;
+        ModuleAnalysisManager MAM;
+
+        PassBuilder PB;
+
+        PB.registerModuleAnalyses(MAM);
+        PB.registerCGSCCAnalyses(CGAM);
+        PB.registerFunctionAnalyses(FAM);
+        PB.registerLoopAnalyses(LAM);
+        PB.crossRegisterProxies(LAM, FAM, CGAM, MAM);
```
Comment (AlexeySachkov) [inline clang/lib/CodeGen/CodeGenAction.cpp:347]: Do we really  need all those things to simply run a single pass?

**Review by AlexeySachkov — COMMENTED** [review]
Tagging @Ruyk for awareness as well

File: clang/lib/CodeGen/CodeGenAction.cpp (hunk 1 above)
Comment (mlychkov) [reply clang/lib/CodeGen/CodeGenAction.cpp]: I reduced pass manager initialization to a minimum required settings.

**Review by AlexeySachkov — APPROVED** [review]

**Review by smanna12 — APPROVED** [review]

**Pull request merged**